### ISSUE TITLE
Voyager doesn't work with cloud = minikube and type = HostPort

### ISSUE THREAD

**Issue by joshk0** [appscode/voyager#272]
I'm trying to use voyager 3.0 on kubernetes 1.5.1 (minikube) with voyager.ingress.appscode.com (extended ingress) in HostPort mode and cloud = minikube.

I find that voyager is able to detect the creation of the TPR, but it does not create the service, configmap, or deployment for the ingress.

Before I go sourcediving, is this at all supported or is cloud = minikube not recommended for any use?

My use case is for people developing against Voyager without needing to push to AWS.

cc: @julianvmodesto

**Comment by sadlil** [comment]
@joshk0 For development purpose voyager [supports `minikube`](https://github.com/appscode/voyager/blob/master/pkg/ingress/parser.go#L24).
If you could provide operator logs we can take a look on this.

**Comment by joshk0** [comment]
The problem was that lbc.Create() was returning an error, but nothing was printing that final error since callers call lbc.Create() without checking for error output.

There were two issues:
* I did not use `kubernetes.io/ingress.class: "voyager"`. But I am using the TPR to create my Ingress. Why does this additionally need to be set?
* I had chosen HostPort style, but did not have node-selector annotation. The documentation does say this is a mandatory annotation, but I can't understand why? A blank nodeSelector is valid in a Kubernetes pod spec.

I worked around it just now by removing the check in createHostPortPods(). I will probably add a silly nodeSelector, something like `beta.kubernetes.io/os=linux`
